salt/modules/virt.py

```diff
@@ -1168,22 +1168,55 @@ def _handle_remote_boot_params(orig_boot):
     """
     saltinst_dir = None
     new_boot = orig_boot.copy()
+    keys = orig_boot.keys()
 
     try:
-        for key in ['kernel', 'initrd']:
-            if check_remote(orig_boot.get(key)):
-                if saltinst_dir is None:
-                    os.makedirs(CACHE_DIR)
-                    saltinst_dir = CACHE_DIR
+        if boot_param_is_valid(keys):
```

Comment:
```suggestion
        if not boot_param_is_valid(keys):
            raise SaltInvocationError("Invalid boot parameter: kernel and initrd need to be both provided")
```

And then you don't change the indentation level of the code after. If the state parameters are wrong, then we need to fail and tell it to the user.